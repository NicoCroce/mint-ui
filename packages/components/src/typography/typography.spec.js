import * as React from "react";
import { render } from "@testing-library/react";
import { Title, Text, Striked} from ".";


describe("Typography tests", () => {
    test("Title renders correctly", () => {
        const { asFragment } = render(<Title text="Heading 1" level="1"  />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("Text styles renders correctly", () => {
        const { asFragment } = render(<Text textStyle="uppercase" textSpace={0.8} text="UPPERCASE Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur" />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("Font weights renders correctly", () => {
        const { asFragment } = render(<Text text="LIGHT Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur" size={20} weight="light" />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("Text Types renders correctly", () => {
        const { asFragment } = render(<Text textType="warning" text="Warning Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur"  weight="light" />);
        expect(asFragment()).toMatchSnapshot();
    });
});