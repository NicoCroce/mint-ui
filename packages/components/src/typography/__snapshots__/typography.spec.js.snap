// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography tests Font weights renders correctly 1`] = `
<DocumentFragment>
  <span
    class="typography__Wrapper-sc-1jpifeu-0 kiqios"
  >
    <span
      class="typography__Content-sc-1jpifeu-1 beRdms"
      size="20"
    >
      LIGHT Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur
    </span>
  </span>
</DocumentFragment>
`;

exports[`Typography tests Text Types renders correctly 1`] = `
<DocumentFragment>
  <span
    class="typography__Wrapper-sc-1jpifeu-0 kiqios"
  >
    <span
      class="typography__Content-sc-1jpifeu-1 bpfwXB"
    >
      Warning Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur
    </span>
  </span>
</DocumentFragment>
`;

exports[`Typography tests Text styles renders correctly 1`] = `
<DocumentFragment>
  <span
    class="typography__Wrapper-sc-1jpifeu-0 kiqios"
  >
    <span
      class="typography__Content-sc-1jpifeu-1 fexyZE"
    >
      UPPERCASE Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur
    </span>
  </span>
</DocumentFragment>
`;

exports[`Typography tests Title renders correctly 1`] = `
<DocumentFragment>
  <span
    class="typography__Wrapper-sc-1jpifeu-0 kiqios"
  >
    <span
      class="typography__Content-sc-1jpifeu-1 etKRG"
    >
      Heading 1
    </span>
  </span>
</DocumentFragment>
`;
